packages/opentelemetry-metrics/src/Metric.ts

```diff
@@ -145,6 +145,15 @@ export class CounterMetric extends Metric<CounterHandle> {
       this._onUpdate
     );
   }
+
+  /**
+   * Adds the given value to the current value. Values cannot be negative.
+   * @param value the value to add.
+   * @param labelSet the canonicalized LabelSet used to associate with this metric's handle.
+   */
+  add(value: number, labelSet: types.LabelSet) {
```

Comment:
Do you think we should include this on the top level API in [types](https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-types/src/metrics/Metric.ts#L71) package.

Reply:
yes, I thought about it, do you think it makes sense to add a `update(value, labelSet)` in types/Metric you linked to? thanks~

Reply:
Maybe `record(value, labelSet)`, if we can't use `add(value, labelSet)` and `set(value, labelSet)`. WDYT?

Reply:
would it be a little bit confused with measure's record function? WDYT?

Reply:
Hmm makes sense. I am ok with current approach for now. Let's wait for others to comment on this.

Reply:
You could add something like this to the Metric type:

```typescript
export interface Metric<T> {
  ...
  
  add: T extends CounterHandle ? CounterHandle["add"] : never;
  set: T extends GaugeHandle ? GaugeHandle["set"] : never;
  record: T extends MeasureHandle ? MeasureHandle["record"] : never;
}
```

Reply:
@dyladan Thanks for your advice. I didn't make your suggestion work.. but it helps me find out Pick interface that enables what we need here. Please take a look and let me know if it works. 😁